contracts/src/StakeTable.sol

```diff
@@ -101,27 +111,48 @@ contract StakeTable is AbstractStakeTable {
         return nodes[_hashBlsKey(blsVK)];
     }
 
+    /// @dev Helper function to avoid code duplication. Contains the core logic of a queue for
+    /// finding the next epoch.
+    /// @param queueType can be Registration or Exit.
+    /// @return Number of the epoch when a user can register/exit.
+    function _nextEpoch(QueueType queueType) private returns (uint64) {
+        Queue storage queue;
+        if (queueType == QueueType.Registration) {
+            queue = registrationQueue;
+        } else {
+            queue = exitQueue;
+        }
+        if (queue.firstAvailableEpoch < currentEpoch() + 1) {
+            queue.firstAvailableEpoch = currentEpoch() + 1;
+            queue.pendingRequests = 1;
+        } else if (queue.pendingRequests >= MAX_CHURN_RATE) {
+            queue.firstAvailableEpoch += 1;
```

Comment:
One problem I see is that, we never increment this `firstAvaialbeEpoch` when we move on to the a new epoch. 

what happens if there's no registration for many epochs, and `currentEpoch > registrationQueue.firstAvailableEpoch`?

Reply:
My impression is that we don't need to update anything in the stake table when the light client increments the epoch number: `currentEpoch` that tracks the latest epoch number from the light client is enough to compute the next registration/exit epoch.

Reply:
See https://github.com/EspressoSystems/espresso-sequencer/pull/803#pullrequestreview-1774819372.